yb_migrate/src/utils/diagnostics.go

```diff
@@ -0,0 +1,143 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/fs"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+//call-home json formats
+var (
+	jsonFilePath    string
+	sendReq         sendRequest
+	jsonBuf         []byte
+	SendDiagnostics bool
+)
+
+type sendRequest struct {
+	MigrationUuid         uuid.UUID `json:"UUID"`                    //done
+	StartTime             string    `json:"start_time"`              //done
+	LastUpdatedTime       string    `json:"last_updated_time"`       //done
+	SourceDBType          string    `json:"source_db_type"`          //done
+	SourceDBVersion       string    `json:"source_db_version"`       //done
+	Issues                string    `json:"issues"`                  //done
+	DBObjects             string    `json:"database_objects"`        //done
+	TargetDBVersion       string    `json:"target_db_version"`       //done
+	NodeCount             int       `json:"node_count"`              //done
+	ParallelJobs          int       `json:"parallel_jobs"`           //done
+	TotalRows             int64     `json:"total_rows"`              //done
+	TotalSize             int64     `json:"total_size"`              //done
+	LargestTableRows      int64     `json:"largest_table_rows"`      //done
+	LargestTableSize      int64     `json:"largest_table_size"`      //done
+	TargetClusterLocation string    `json:"target_cluster_location"` //ask on slack channels for this (answer given is insufficient)
+	TargetDBCores         int       `json:"target_db_cores"`         //unknown for now
+	SourceCloudDBType     string    `json:"source_cloud_type"`       //unknown for now
+}
+
+//Fill in primary-key based fields, if needed
+func InitJSON(exportdir string) {
+	var err error
+	jsonFilePath = filepath.Join(exportdir, "metainfo", "diagnostics.json")
+	_, err = os.OpenFile(jsonFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		ErrExit("Error while creating/opening diagnostics.json file: %v", err)
+	}
+	jsonBuf, err = os.ReadFile(jsonFilePath)
```

Comment:
Option1:
as you have already done `os.OpenFile()`, use file variable returned and create reader for reading from the file as this `ReadFile` function will again internally open the file.

```
reader := bufio.NewReader(file) 
or 
scanner := bufio.NewScanner(file)
```

or 
Option2: why are doing `OpenFile()` to check errors, if there is any error `ReadFile()` should report that right?

Reply:
1. Will it suffice to immediately close the file? I was searching for something similar to `touch`, where the file is created if not exists, but not truncated if it does exist
2. I am not using OpenFile() to check errors, there are different possibilities of errors occurring in each scenario, which I am catching in the code

Reply:
1. os.Create()
2. but isn't that handled/reported by ReadFile() itself?